Firestore/core/src/firebase/firestore/model/unknown_document.cc

```diff
@@ -23,20 +23,26 @@ namespace firebase {
 namespace firestore {
 namespace model {
 
+static_assert(
+    sizeof(MaybeDocument) == sizeof(UnknownDocument),
+    "UnknownDocument may not have additional members (everything goes in Rep)");
+
 class UnknownDocument::Rep : public MaybeDocument::Rep {
  public:
   Rep(DocumentKey key, SnapshotVersion version)
       : MaybeDocument::Rep(Type::UnknownDocument, std::move(key), version) {
   }
 
   bool has_pending_writes() const override {
+    // Unknown documents can only exist because of a logical inconsistency
+    // between the server successfully committing a mutation that our local
```

Comment:
Nit: `between` is missing the counterpart; consider either tweaking the phrasing to something like "between the server successfully committing a mutation _and_ our local cache _believing it_ should not apply", or just replacing between with something like "of" ("the logical inconsistency of the server...").

Reply:
Took the former advice.